akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -128,7 +128,10 @@ class LightArrayRevolverScheduler(config: Config, log: LoggingAdapter, threadFac
         }
       }
 
-      @tailrec final def cancel(): Boolean = {
+      // TODO DOTTY
+      // Cannot rewrite recursive call: it targets a supertype
```

Comment:
I'm puzzled by this error message, is it the expected behavior @nicolasstucki, @smarter ?

Reply:
The error message refers to`c.cancel()` which is an overridden method of the current `cancel()` method, dotty correctly tells you that it cannot rewrite it, scala 2 doesn't say anything here, it's not a bug, just a difference of what `@tailrec` checks.

Reply:
done in #29939